mongo-fixes/update-topic-ids/update-topic-ids.js

```diff
@@ -0,0 +1,103 @@
+// update-topic-id.js
+//
+// Update all topic ids to use a new nano id
+
+const collection = 'topics'
+const idSize = 4
+const idAlphabet = '123456789'
+
+if (typeof(cfg) == "undefined") {
+    // default, but can be changed on command-line, see README
+    cfg = {
+        verbose:  false,    // do a find first, show what would be changed
+        update:   true      // set to false to avoid updates
+    }
+}
+
+//////////////////////////
+
+var idMap = new Object() // Store oldId => newId
+var allNewIds = []  // Store all new ids to check for clashes
+function getNewId(oldId) {
+    if (idMap[oldId] == null) {
+        do {
+            newId = makeId()
+        } while (isUsedId(newId))
+        allNewIds.push(newId)
+        idMap[oldId] = newId
+        print(oldId + " becomes " + newId)
+    }
+    return idMap[oldId]
+}
+
+function isUsedId(id) {
+    for (var i in allNewIds) {
+        if (i == id) {
+            return true
+        }
+    }
+    return false
+}
+
+function makeId() {
+    var result = ''
+    for (var i = 0; i < idSize; i++) {
+      result += idAlphabet.charAt(Math.floor(Math.random() * idAlphabet.length))
+    }
+    return result
+}
+
+function updateTopicDocument(topic) {
+    var oldId = topic.id
+    if (oldId != 'topic_root') {
+        var newId = getNewId(topic.id)
+
+        topic.id = newId
+        topic.next.id = newId
+        topic.current.id = newId
```

Comment:
Check for null pointer on topic.next and topic.current?
I had some local topics that did not have a next or current. It's quite possible my topics were malformed, so just bringing this to your attention, in case it's valid for a topic to have one or more of these pointers to be nil